src/app/auth/services/auth.service.ts

```diff
@@ -213,31 +206,14 @@ export const changePassword = async (newPassword: string, currentPassword: strin
     });
 };
 
-export const userHas2FAStored = async (): Promise<{
-  has2fa: boolean;
-  data: { hasKeys: boolean; sKey: string; tfa: boolean };
-}> => {
-  const response = await fetch(`${process.env.REACT_APP_API_URL}/api/login`, {
-    method: 'POST',
-    headers: httpService.getHeaders(true, false),
-    body: JSON.stringify({ email: JSON.parse(localStorage.xUser).email }),
-  });
-  const data = await response.json();
-
-  return { has2fa: typeof data.tfa === 'boolean', data };
+export const userHas2FAStored = async (): Promise<SecurityDetails> => {
+  const email = localStorageService.getUser()?.email;
+  const authClient = createAuthClient();
+  return await authClient.securityDetails(<string>email);
```

Comment:
you can remove `async` and `await` without problems, returning a promise is the same as `async` and returning the awaited value